ee/clickhouse/migrations/0027_persons_and_groups_on_events.py

```diff
@@ -0,0 +1,35 @@
+from infi.clickhouse_orm import migrations
+
+from ee.clickhouse.replication.utils import clickhouse_is_replicated
+from ee.clickhouse.sql.events import EVENTS_TABLE_JSON_MV_SQL, KAFKA_EVENTS_TABLE_JSON_SQL
+from posthog.client import sync_execute
+from posthog.settings import CLICKHOUSE_CLUSTER
+
+ADD_COLUMNS_BASE_SQL = """
+ALTER TABLE {table}
+ON CLUSTER '{cluster}'
+ADD COLUMN IF NOT EXISTS person_id UUID,
+ADD COLUMN IF NOT EXISTS person_properties VARCHAR,
+ADD COLUMN IF NOT EXISTS group0_properties VARCHAR,
+ADD COLUMN IF NOT EXISTS group1_properties VARCHAR,
+ADD COLUMN IF NOT EXISTS group2_properties VARCHAR,
+ADD COLUMN IF NOT EXISTS group3_properties VARCHAR,
+ADD COLUMN IF NOT EXISTS group4_properties VARCHAR
```

Comment:
I'd replace `VARCHAR` with `String` here just because `VARCHAR` is just an alias to `String`
https://clickhouse.com/docs/en/operations/system-tables/data_type_families#system_tables-data_type_families

Reply:
Aware of this, but was using varchar to keep things consistent in this table. See the current definition:


```python
EVENTS_TABLE_BASE_SQL = """
CREATE TABLE IF NOT EXISTS {table_name} ON CLUSTER '{cluster}'
(
    uuid UUID,
    event VARCHAR,
    properties VARCHAR,
    timestamp DateTime64(6, 'UTC'),
    team_id Int64,
    distinct_id VARCHAR,
    elements_chain VARCHAR,
    created_at DateTime64(6, 'UTC'),
    person_id UUID,
    person_properties VARCHAR,
    group0_properties VARCHAR,
    group1_properties VARCHAR,
    group2_properties VARCHAR,
    group3_properties VARCHAR,
    group4_properties VARCHAR
    {materialized_columns}
    {extra_fields}
) ENGINE = {engine}
"""
```